File: api/shared/models/creditTransaction.ts
export interface CreditTransaction {
  id: string;
  userId: string;
  amount: number;
  type: 'PURCHASE' | 'DEDUCTION' | 'REFUND';
  description: string;
  createdAt: string;
  reference?: string;
}

export interface CreditTransactionDTO {
  userId: string;
  amount: number;
  type: 'PURCHASE' | 'DEDUCTION' | 'REFUND';
  description: string;
  reference?: string;
}

File: api/shared/models/credit_transaction.py
from typing import Optional, Literal
from pydantic import BaseModel

class CreditTransaction(BaseModel):
  id: str
  user_id: str
  amount: int
  type: Literal['PURCHASE', 'DEDUCTION', 'REFUND']
  description: str
  created_at: str
  reference: Optional[str] = None

class CreditTransactionDTO(BaseModel):
  user_id: str
  amount: int
  type: Literal['PURCHASE', 'DEDUCTION', 'REFUND']
  description: str
  reference: Optional[str] = None